File: solutions/c/darts/1/darts.c
#include "darts.h"

uint8_t score(coordinate_t coord) {
   float dist = coord.x * coord.x + coord.y * coord.y;
   if (dist > 100)
      return 0;
   if (dist > 25)
      return 1;
   if (dist > 1)
      return 5;
   return 10;
}

File: solutions/elixir/darts/1/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score({x, y}) do
    distance = x ** 2 + y ** 2

    cond do
      distance > 100 -> 0
      distance > 25 -> 1
      distance > 1 -> 5
      true -> 10
    end
  end
end
